File: classic_gym/classic_gym/report/member_fitness_journey/member_fitness_journey.js
// Copyright (c) 2024, Polito and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Member Fitness Journey"] = {
	"filters": [
		{
		"fieldname" : "member_name",
		"label":__('Member Name'),
		"fieldtype": 'Link',
		"options":"Gym Member",
		 "width": 200,
		"reqd": 0

		},
		{
				
			"fieldname": "age",
            "label": __("Age"),
            "fieldtype": "Data",
            "width": 200,
            "reqd": 0
		
				},
		{
			"fieldname" : "member_weight",
			"label":__('Member Weight'),
			"fieldtype": 'Link',
			"options":"Gym Member Weight",
			"width": 200,
			"reqd": 0
	
			},
		{
				
			"fieldname": "date",
            "label": __("Date"),
            "fieldtype": "Date",
            "width": 200,
            "reqd": 0
		
				}
	]
};

File: classic_gym/classic_gym/report/member_fitness_journey/member_fitness_journey.py
# Copyright (c) 2024, Polito and contributors
# For license information, please see license.txt

import frappe


def execute(filters=None):
    columns = get_columns()
    data = get_data(filters)
    return columns, data


def get_data(filters):
    conditions = "1=1"
    if filters.get("member_name"):
        conditions += f" AND gm.email='{filters.get('member_name')}'"

    SQL = f"""
        SELECT 
            gm.email,
            gm.age,
            mw.weight,
            mw.date
        FROM
            `tabGym Member` AS gm
        JOIN 
            `tabGym Member Weight` AS mw
        ON
            gm.email = mw.member_email
        WHERE
            {conditions}
    """
    data = frappe.db.sql(SQL)
    return data


def get_columns():
    return [
        "Member Name:Link/Gym Member:200",
        "Age:Data:200",
        "Member Weight:Link/Gym Member Weight:250",
        "Date:Date:200"
        
    ]
